### ISSUE TITLE
Unable to explicitly bind mutation error type

### ISSUE THREAD

**Issue by glen-84** [ChilliCream/graphql-platform#6761]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

When setting the default binding behaviour to `Explicit`, I see the following error:

> 1. Error while building type HotChocolate.Types.ErrorObjectType`1[[hc_test_mutation_errors.Types.MyError, hc-test-mutation-errors, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]. The runtime type hc_test_mutation_errors.Types.MyError does not defined a property 'Message'

I've tried to explicitly bind the type in `MyErrorType.cs`, but it doesn't appear to be working.

`MyError` is used in the `MutationResult` in `Mutation.cs`.

### Steps to reproduce

https://github.com/glen-84/hc-test-mutation-errors

### Relevant log output

```shell
1. Error while building type HotChocolate.Types.ErrorObjectType`1[[hc_test_mutation_errors.Types.MyError, hc-test-mutation-errors, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]]. The runtime type hc_test_mutation_errors.Types.MyError does not defined a property `Message`
 (HotChocolate.Types.ErrorObjectType<hc_test_mutation_errors.Types.MyError>)

   at HotChocolate.Configuration.TypeInitializer.DiscoverTypes()
   at HotChocolate.Configuration.TypeInitializer.Initialize()
   at HotChocolate.SchemaBuilder.Setup.InitializeTypes(SchemaBuilder builder, IDescriptorContext context, IReadOnlyList`1 types)
   at HotChocolate.SchemaBuilder.Setup.Create(SchemaBuilder builder, LazySchema lazySchema, IDescriptorContext context)
   at HotChocolate.SchemaBuilder.Create(IDescriptorContext context)
   at HotChocolate.SchemaBuilder.HotChocolate.ISchemaBuilder.Create(IDescriptorContext context)
   at HotChocolate.Execution.RequestExecutorResolver.CreateSchemaAsync(ConfigurationContext context, RequestExecutorSetup setup, RequestExecutorOptions executorOptions, IServiceProvider schemaServices, TypeModuleChangeMonitor typeModuleChangeMonitor, CancellationToken cancellationToken)
   at HotChocolate.Execution.RequestExecutorResolver.CreateSchemaServicesAsync(ConfigurationContext context, RequestExecutorSetup setup, CancellationToken cancellationToken)
   at HotChocolate.Execution.RequestExecutorResolver.GetRequestExecutorNoLockAsync(String schemaName, CancellationToken cancellationToken)
   at HotChocolate.Execution.RequestExecutorResolver.GetRequestExecutorAsync(String schemaName, CancellationToken cancellationToken)
   at HotChocolate.AspNetCore.Warmup.ExecutorWarmupService.ExecuteAsync(CancellationToken stoppingToken)
   at Microsoft.Extensions.Hosting.Internal.Host.<StartAsync>b__15_1(IHostedService service, CancellationToken token)
   at Microsoft.Extensions.Hosting.Internal.Host.ForeachService[T](IEnumerable`1 services, CancellationToken token, Boolean concurrent, Boolean abortOnFirstException, List`1 exceptions, Func`3 operation)
   at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.Run(IHost host)
   at Microsoft.Extensions.Hosting.HostBuilderExtensions.RunWithGraphQLCommands(IHost host, String[] args)
   at Program.<Main>$(String[] args) in /home/glen/projects/hc-test-mutation-errors/Program.cs:line 14
```

### Version

13.7.0

**Comment by glen-84** [comment]
Here's another example where I'm trying to use a custom error interface with explicit binding:

https://github.com/glen-84/hc-error-explicit-binding

> 1. The field `code` must be implement by object type `TestError`.

**Comment by glen-84** [comment]
Fixed by #7548.
